module-services/service-desktop/Sync.cpp

```diff
@@ -0,0 +1,205 @@
+﻿// Copyright (c) 2017-2022, Mudita Sp. z.o.o. All rights reserved.
+// For licensing, see https://github.com/mudita/MuditaOS/LICENSE.md
+
+#include <service-desktop/Sync.hpp>
+#include <service-db/DBServiceAPI.hpp>
+#include <purefs/filesystem_paths.hpp>
+
+#include <microtar.hpp>
+
+namespace sys
+{
+    class Service;
+} // namespace sys
+
+static bool isValidDirentry(const std::filesystem::directory_entry &direntry)
+{
+    return direntry.path() != "." && direntry.path() != ".." && direntry.path() != "...";
+}
+
+Sync::CompletionCode Sync::PrepareSyncPackage(sys::Service *ownerService, std::filesystem::path &path)
+{
+    assert(ownerService != nullptr);
+    LOG_INFO("Sync package preparation started...");
+
+    if (!Sync::RemoveSyncDir(path)) {
```

Comment:
An error will be returned as a response to the sync package request, but yes - a new request can be received then, to start the sync process again.

Reply:
But - if I understand the code correctly - the same directory will be passed again to that function, and most probably it fill fail again, right? If we aren't able to remove the directory for some reason, we probably won't be able to do so in the next request.
Can we work around such an error somehow? E.g. pass another path?

Reply:
No, it will be different dir, because `taskId` will change: `syncStatus.tempDir = purefs::dir::getTemporaryPath() / syncStatus.taskId;`
If this function fails, we have a filesystem error and we cannot do much more here.